modules/kms-keyring/src/kms_keyring.ts

```diff
@@ -259,7 +259,9 @@ export function KmsKeyringClass<
            * If the caller does not have access they may have access
            * through another Keyring.
            */
-          cmkErrors.push(e)
+          if (e instanceof Error) {
+            cmkErrors.push(e)
+          }
```

Comment:
All the changes involving `catch(<var>)... <do something with var>` are caused by [Using `unkown` in Catch Variables](https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#using-unknown-in-catch-variables).
There are two alternatives to this change:
1. `--useUnknownInCatchVariables`, which will disable this feature globally.
2. Export an interface that can handle `Error | any | unknown` and have `cmkErrors` use that.

Reply:
2 is the better option. Because it is true the JS can throw anything, and many of the checks that you have added have subtle impacts that we want to avoid. e.g. continuing after specific types of errors :(